compiler/rustc_codegen_llvm/src/gotoc/cbmc/goto_program/expr.rs

```diff
@@ -698,6 +698,27 @@ impl Expr {
         Expr::struct_expr_with_explicit_padding(typ, fields, values)
     }
 
+    /// Struct initializer  
+    /// `struct foo the_foo = >>> {field1, padding2, field3, ... } <<<`
+    /// Note that padding fields should be explicitiy given.
+    pub fn struct_expr_from_padded_values(
+        typ: Type,
+        mut values: Vec<Expr>,
+        symbol_table: &SymbolTable,
+    ) -> Self {
+        assert!(typ.is_struct_tag(), "Error in struct_expr\n\t{:?}\n\t{:?}", typ, values);
+        let fields = symbol_table.lookup_fields_in_type(&typ).unwrap();
+        assert_eq!(fields.len(), values.len(), "Error in struct_expr\n\t{:?}\n\t{:?}", typ, values);
```

Comment:
These two error messages are equal but the checks are different. Use different messages here.

Reply:
Fixed!